src/chrome/cdtpDebuggee/eventsProviders/cdtpExceptionThrownEventsProvider.ts

```diff
@@ -57,12 +58,15 @@ export class CDTPExceptionThrownEventsProvider extends CDTPEventsEmitterDiagnost
     }
 
     private async toExceptionDetails(exceptionDetails: CDTP.Runtime.ExceptionDetails): Promise<IExceptionDetails> {
+        const scriptId = exceptionDetails.scriptId;
+
         return {
             exceptionId: exceptionDetails.exceptionId,
             text: exceptionDetails.text,
             lineNumber: exceptionDetails.lineNumber,
             columnNumber: exceptionDetails.columnNumber,
-            script: isNotEmpty(exceptionDetails.scriptId) ? await this._scriptsRegistry.getScriptByCdtpId(exceptionDetails.scriptId) : undefined,
+            // We've seen scriptId contain invalid script ids. If that happens, we just ignore it
+            script: isNotEmpty(scriptId) ? await asyncUndefinedOnFailure(() => this._scriptsRegistry.getScriptByCdtpId(scriptId)) : undefined,
```

Comment:
Suggestion: move this logic out of the declaration of the return object and create a variable. It's just a lot of stuff happening on one line in the middle of a declaration.

Reply:
Done